src/compile/data/bin.ts

```diff
@@ -123,10 +125,16 @@ export class BinNode extends DataFlowNode {
    * Merge bin nodes. This method either integrates the bin config from the other node
    * or if this node already has a bin config, renames the corresponding signal in the model.
    */
-  public merge(other: BinNode, model: Model) {
+  public merge(other: BinNode, renameSignal: (s1: string, s2: string) => void) {
     for (const key of keys(other.bins)) {
       if (key in this.bins) {
-        model.renameSignal(other.bins[key].signal, this.bins[key].signal);
+        renameSignal(other.bins[key].signal, this.bins[key].signal);
+        // Ensure that we don't have duplicate names for signal pairs
+        const uniqueAs: Dict<[string, string]> = {};
+        for (const as of [...this.bins[key].as, ...other.bins[key].as]) {
+          uniqueAs[hash(as)] = as;
+        }
+        this.bins[key].as = vals(uniqueAs);
```

Comment:
A slightly simpler implementation is to just create an array with `as` and then using `util.unique(asArray, util.hash)`.

Reply:
That seems reasonable. I'll give it a try.